caffe2/operators/hip/conv_op_miopen.hip

```diff
@@ -252,48 +253,101 @@ bool MIOPENConvOp::DoRunWithType() {
+        MIOPEN_ENFORCE(miopenInitConvolutionDescriptor(
+            conv_desc_,
+            mode_,
+            pad_t(),
+            pad_l(),
+            stride_h(),
+            stride_w(),
+            dilation_h(),
+            dilation_w()));
+      } else {
+        int spatial_dim = 3;
+        MIOPEN_ENFORCE(miopenInitConvolutionNdDescriptor(
+            conv_desc_,
+            spatial_dim,
+            pads_.data(),
+            stride_.data(),
+            dilation_.data(),
+            mode_));
+      }
 
       MIOPEN_ENFORCE(miopenSetConvolutionGroupCount(conv_desc_, group_));
 
-      MIOPEN_ENFORCE(miopenSet4dTensorDescriptor(
-          weight_desc_,
-          miopenTypeWrapper<T_W>::type,
-          M,
-          C / group_,
-          kernel_h(),
-          kernel_w()));
+      if (mio_weight_dims_.size() == 4) {
+        MIOPEN_ENFORCE(miopenSet4dTensorDescriptor(
+            weight_desc_,
+            miopenTypeWrapper<T_W>::type,
+            M,
+            C / group_,
+            kernel_h(),
+            kernel_w()));
+      } else {
+        vector<int> weight_dims (begin(mio_weight_dims_), end(mio_weight_dims_));
+        MIOPEN_ENFORCE(miopenSetTensorDescriptor(
+            weight_desc_,
+            miopenTypeWrapper<T_W>::type,
+            mio_weight_dims_.size(),
+            weight_dims.data(),
+            nullptr));
+      }
     }
 
-    MIOPEN_ENFORCE(miopenGetConvolutionForwardOutputDim(
-        conv_desc_,
-        bottom_desc_,
-        weight_desc_,
-        &N_out,
-        &C_out,
-        &H_out,
-        &W_out));
+    
+    mio_output_dims_ = Y->sizes().vec();
+    int output_dim_size = mio_input_dims_.size(); 
+    vector<int> output_dims (begin(mio_output_dims_), end(mio_output_dims_));
+    if (mio_input_dims_.size() == 4) {
+      MIOPEN_ENFORCE(miopenGetConvolutionForwardOutputDim(
+          conv_desc_,
+          bottom_desc_,
+          weight_desc_,
+          &N_out,
+          &C_out,
+          &H_out,
+          &W_out));
+    } 
+    else {
+      MIOPEN_ENFORCE(miopenGetConvolutionNdForwardOutputDim(
+          conv_desc_,
+          bottom_desc_,
+          weight_desc_,
+          &output_dim_size,
+          output_dims.data()));
+    }
 
-    MIOPEN_ENFORCE(miopenSet4dTensorDescriptor(
-        top_desc_, miopenTypeWrapper<T_X>::type, N_out, C_out, H_out, W_out));
+    if (mio_input_dims_.size() == 4) {
+      MIOPEN_ENFORCE(miopenSet4dTensorDescriptor(
+          top_desc_, miopenTypeWrapper<T_X>::type, N_out, C_out, H_out, W_out));
+    } else {
+      MIOPEN_ENFORCE(miopenSetTensorDescriptor(
+          top_desc_, miopenTypeWrapper<T_X>::type, output_dims.size(), output_dims.data(), nullptr));
+    }
 
     if (InputSize() == 3) {
-      MIOPEN_ENFORCE(miopenSet4dTensorDescriptor(
-          bias_desc_, miopenTypeWrapper<T_B>::type, 1, M, 1, 1));
+      if (mio_input_dims_.size() == 4) {
+        MIOPEN_ENFORCE(miopenSet4dTensorDescriptor(
```

Comment:
General question: could we not use the generic miopenSetTensorDescriptor also for the 4D case and eschew these if/else here?

Reply:
Haven't considered about refactoring the existing code, but yes 4D case can also be part of miopenSetTensorDescriptor. Let me clean up the code.